crates/docs/src/utils.rs

```diff
@@ -27,15 +29,45 @@ pub fn assert_valid_snippet(condition: bool, snippet: &Snippet, err_message: &st
     );
 }
 
-pub fn print_success_message(snippets_len: usize, tool_name: &str) {
-    println!("Successfully validated {snippets_len} {tool_name} docs snippets");
+pub fn print_snippets_validation_summary(snippets: &[Snippet], tool_name: &str) {
+    let validated_snippets_count = snippets
+        .iter()
+        .filter(|snippet| !snippet.config.ignored)
+        .count();
+    let ignored_snippets_count = snippets.len() - validated_snippets_count;
+
+    println!("Finished validation of {tool_name} docs snippets\nValidated: {validated_snippets_count}, Ignored: {ignored_snippets_count}");
 }
 
-pub fn print_skipped_snippet_message(snippet: &Snippet) {
+pub fn print_ignored_snippet_message(snippet: &Snippet) {
     println!(
-        "Skipped validation of {} snippet in the docs in file: {} at line {}",
+        "Ignoring {} docs snippet, file: {}:{}:1",
         snippet.snippet_type.as_str(),
         snippet.file_path,
         snippet.line_start,
     );
 }
+
+fn get_canonical_path(relative_path: &str) -> Result<String, Box<dyn std::error::Error>> {
+    Ok(Utf8PathBuf::from_str(relative_path)?
+        .canonicalize_utf8()?
+        .to_string())
```

Comment:
You can do `map_err(|e| anyhow!(...))` and return an anyhow Error instead of `Box<dyn std::error::Error>`